lib/js/src/manager/screen/choiceset/_ChoiceSetManagerBase.js

```diff
@@ -594,20 +621,81 @@ class _ChoiceSetManagerBase extends _SubManagerBase {
             .setKeypressMode(KeypressMode.RESEND_CURRENT_ENTRY);
     }
 
+    /**
+     * Clones a list of choice cells
+     * @param {ChoiceCell[]} originalList - A list of choice cells to be cloned
+     * @returns {ChoiceCell[]} - The cloned
+     */
+    _cloneChoiceCellList (originalList) {
+        if (!Array.isArray(originalList)) {
+            return null;
+        }
+        return originalList.map((choiceCell) => choiceCell.clone());
+    }
+
     /**
      * Modifies the choices names depending on SDL version
      * @param {ChoiceCell[]} choices - The first list of choices
      * @returns {ChoiceCell[]} - A deep copy of the name modified choices
      */
     _getChoicesToBeUploadedWithArray (choices) {
+        const choicesClone = this._cloneChoiceCellList(choices);
         // If we're running on a connection < RPC 7.1, we need to de-duplicate cells because presenting them will fail if we have the same cell primary text.
         if (choices !== null && this._lifecycleManager.getSdlMsgVersion() !== null
                 && (this._lifecycleManager.getSdlMsgVersion().getMajorVersion() < 7
                 || (this._lifecycleManager.getSdlMsgVersion().getMajorVersion() === 7 && this._lifecycleManager.getSdlMsgVersion().getMinorVersion() === 0))) {
             // version if 7.0.0 or lower
-            this._addUniqueNamesToCells(choices);
+            this._addUniqueNamesToCells(choicesClone);
+        } else {
+            const strippedCellsClone = this._removeUnusedProperties(choicesClone);
+            this._addUniqueNamesBasedOnStrippedCells(strippedCellsClone, choicesClone);
+        }
+        this._preloadedChoices.forEach((preloadedChoice) => {
+            _ArrayTools.arrayRemove(choicesClone, preloadedChoice);
```

Comment:
_ArrayTools.arrayRemove's method does not mutate the first argument, so choicesClone will not be changed by the end. Clever use of `reduce` can resolve this:

`return this._preloadedChoices.reduce((filteredChoices, choice) => _ArrayTools.arrayRemove(filteredChoices, choice), choicesClone);`